src/ray/gcs/gcs_server/pubsub_handler.h

```diff
@@ -51,6 +54,10 @@ class InternalPubSubHandler : public rpc::InternalPubSubHandler {
   void RemoveSubscriberFrom(const std::string &sender_id);
 
  private:
+  /// Whether to handshake client with server token.
+  bool const token_handshake_;
+  /// GCS server token to pass to subscribers.
+  std::string const &gcs_token_;
```

Comment:
I think we don't need to put the server token here. It should be in the RPC layer, not application layer I think.